chessboard/olegbet/chessboard.go

```diff
@@ -0,0 +1,54 @@
+package chessboard
+
+// Declare a type named Rank which stores if a square is occupied by a piece - this will be a slice of bools
+
+type Rank []bool
+
+// Declare a type named Chessboard contains a map of eight Ranks, accessed with values from "A" to "H"
+
+type Chessboard map[string]Rank
+
+// CountInRank returns how many squares are occupied in the chessboard,
+// within the given rank
+func CountInRank(cb Chessboard, rank string) (count int) {
+	r := cb[rank]
+	for _, v := range r {
+		if v {
+			count++
+		}
+	}
+	return
+}
+
+// CountInFile returns how many squares are occupied in the chessboard,
+// within the given file
+func CountInFile(cb Chessboard, file int) (count int) {
+	if file >= 0 && file <= len(cb["A"]) {
```

Comment:
I think it's better to have an earlier return here so you don't encapsulate the main login within the check.
```go
if file >= 0 && file <= len(cb["A"]) {
    return 0
}

...
```